### PR TITLE
chore: Update compliance tools to the next generation

### PR DESCRIPTION
**Pull request by DaveTryon**
#### Details

The compliance tools that we use are being revised, as shown in the following warnings (adapted from this morning's signed build)

Job | Task | Warning
--- | --- | ---
ComplianceDebug | Run AutoApplicability | This tool is no longer supported. Please remove this build task from your build definition.
ComplianceDebug | Run CredScan | Please upgrade the version of this build task in your build to major version: 3.
ComplianceDebug | Run Roslyn analyzers | Please upgrade the version of this build task in your build to major version: 3.
ComplianceDebug | Run PoliCheck | Please upgrade the version of this build task in your build to major version: 2.
ComplianceDebug | Create Security Analysis Report (CredScan, RoslynAnalyzers, and PoliCheck) | Please upgrade the version of this build task in your build to major version: 2.
ComplianceDebug | Publish Security Analysis Logs (CredScan, RoslynAnalyzers, and PoliCheck) | Please upgrade the version of this build task in your build to major version: 3.
ComplianceDebug | Post Analysis (CredScan, RoslynAnalyzers, and PoliCheck) | Please upgrade the version of this build task in your build to major version: 2.
ComplianceDebug | TSA upload (CredScan, RoslynAnalyzers, and PoliCheck) | Please upgrade the version of this build task in your build to major version: 2.
SignedRelease | Run BinSkim | Please upgrade the version of this build task in your build to major version: 4.
SignedRelease | Create Security Analysis Report (BinSkim) | Please upgrade the version of this build task in your build to major version: 2.
SignedRelease | Publish Security Analysis Logs (BinSkim) | Please upgrade the version of this build task in your build to major version: 3.
SignedRelease | Post Analysis (BinSkim) | Please upgrade the version of this build task in your build to major version: 2.
SignedRelease | TSA upload (BinSkim) | lease upgrade the version of this build task in your build to major version: 2.

This PR makes the specified changes, with some variations:
1. The RoslynAnalyzers task also includes changes to reflect the use of VS17, as well as changing an input value that is case-sensitive in the new tool version.
2. The new TSA uploader task requires its input via a config file instead of via command line parameters. Some of this information (emails, etc.) should be protected and is stored as pipeline variables. We build `tsa.config` on the fly. This script is intended to be easily modified for use in our other pipelines.
3. The new CredScan task default to `sarif` upload format, which doesn't get picked up by the TSA uploader. We need to specify the `pre` output format for consumption by the TSA uploader.

##### Motivation

The current version of security tools will soon be deprecated and we need to keep this functionality for compliance reasons.

#### Pull request checklist

- [ ] Run through of all [test scenarios](https://github.com/Microsoft/accessibility-insights-windows/blob/main/docs/Scenarios.md) completed?
- [ ] Does this address an existing issue? If yes, Issue# - 
- [ ] Includes UI changes?
  - [ ] Run the production version of Accessibility Insights for Windows against a version with changes.
  - [ ] Attach any screenshots / GIF's that are applicable.

> Note: After the PR has been created, certain checks will be kicked off. All of these checks must pass before a merge.

### CODE REVIEW

File: build/signedbuild.yml [hunk 1]
```diff
@@ -139,69 +139,65 @@ jobs:
...
-      PoliCheck: true
 
-  - task: securedevelopmentteam.vss-secure-development-tools.build-task-publishsecurityanalysislogs.PublishSecurityAnalysisLogs@2
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-publishsecurityanalysislogs.PublishSecurityAnalysisLogs@3
     displayName: 'Publish Security Analysis Logs (CredScan, RoslynAnalyzers, and PoliCheck)'
 
-  - task: securedevelopmentteam.vss-secure-development-tools.build-task-postanalysis.PostAnalysis@1
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-postanalysis.PostAnalysis@2
     displayName: 'Post Analysis (CredScan, RoslynAnalyzers, and PoliCheck)'
     inputs:
```
Comment (DaveTryon) [inline build/signedbuild.yml:173]: These parameters have changed in V2 and no longer do anything useful. I'll remove them before merging

File: build/signedbuild.yml [hunk 2]
```diff
@@ -281,26 +277,26 @@ jobs:
...
+      AnalyzeTargetGlob: "src\\AccessibilityInsights\\bin\\Release\\net48\\*.exe;\
+                          src\\AccessibilityInsights\\bin\\Release\\net48\\AccessibilityInsights.*.dll;\
+                          src\\AccessibilityInsights.VersionSwitcher\\bin\\Release\\net48\\*.exe;\
+                          src\\AccessibilityInsights.VersionSwitcher\\bin\\Release\\net48\\AccessibilityInsights.*.dll;\
+                          src\\Manifest\\bin\\Release\\net48\\AccessibilityInsights.*.dll;"
 
-  - task: securedevelopmentteam.vss-secure-development-tools.build-task-report.SdtReport@1
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-report.SdtReport@2
     displayName: 'Create Security Analysis Report (BinSkim)'
     inputs:
```
Comment (DaveTryon) [inline build/signedbuild.yml:293]: TODO: Remove the input parameters (they've changed and we're just using the defaults

File: build/signedbuild.yml [hunk 3]
```diff
@@ -281,26 +277,26 @@ jobs:
...
       BinSkim: true
 
-  - task: securedevelopmentteam.vss-secure-development-tools.build-task-publishsecurityanalysislogs.PublishSecurityAnalysisLogs@2
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-publishsecurityanalysislogs.PublishSecurityAnalysisLogs@3
     displayName: 'Publish Security Analysis Logs (BinSkim)'
 
-  - task: securedevelopmentteam.vss-secure-development-tools.build-task-postanalysis.PostAnalysis@1
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-postanalysis.PostAnalysis@2
     displayName: 'Post Analysis (BinSkim)'
     inputs:
```
Comment (DaveTryon) [inline build/signedbuild.yml:301]: TODO: Remove the input parameters (they've changed and we're just using the defaults

File: build/signedbuild.yml [hunk 4]
```diff
@@ -139,69 +139,65 @@ jobs:
...
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-policheck.PoliCheck@2
     displayName: 'Run PoliCheck'
     inputs:
       targetType: F
     continueOnError: true
 
-  - task: securedevelopmentteam.vss-secure-development-tools.build-task-report.SdtReport@1
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-report.SdtReport@2
     displayName: 'Create Security Analysis Report (CredScan, RoslynAnalyzers, and PoliCheck)'
-    inputs:
```
Comment (DaveTryon) [inline build/signedbuild.yml:171]: Default behavior is to use all available tools, which is exactly what we want

File: build/signedbuild.yml [hunk 5]
```diff
@@ -281,29 +273,25 @@ jobs:
...
+      AnalyzeTargetGlob: "src\\AccessibilityInsights\\bin\\Release\\net48\\*.exe;\
+                          src\\AccessibilityInsights\\bin\\Release\\net48\\AccessibilityInsights.*.dll;\
+                          src\\AccessibilityInsights.VersionSwitcher\\bin\\Release\\net48\\*.exe;\
+                          src\\AccessibilityInsights.VersionSwitcher\\bin\\Release\\net48\\AccessibilityInsights.*.dll;\
+                          src\\Manifest\\bin\\Release\\net48\\AccessibilityInsights.*.dll;"
 
-  - task: securedevelopmentteam.vss-secure-development-tools.build-task-report.SdtReport@1
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-report.SdtReport@2
     displayName: 'Create Security Analysis Report (BinSkim)'
-    inputs:
```
Comment (DaveTryon) [inline build/signedbuild.yml:297]: Default behavior is to use all available tool results, which is exactly what we want

File: build/signedbuild.yml [hunk 6]
```diff
@@ -281,29 +273,25 @@ jobs:
...
-      BinSkim: true
 
-  - task: securedevelopmentteam.vss-secure-development-tools.build-task-publishsecurityanalysislogs.PublishSecurityAnalysisLogs@2
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-publishsecurityanalysislogs.PublishSecurityAnalysisLogs@3
     displayName: 'Publish Security Analysis Logs (BinSkim)'
 
-  - task: securedevelopmentteam.vss-secure-development-tools.build-task-postanalysis.PostAnalysis@1
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-postanalysis.PostAnalysis@2
     displayName: 'Post Analysis (BinSkim)'
-    inputs:
```
Comment (DaveTryon) [inline build/signedbuild.yml:305]: Default behavior is to use all available tool results, which is exactly what we want

File: tools/scripts/pipeline/build/create-tsa-config.ps1 [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+The email alias[es] that will be used for TSA notifications, as a semicolon delimited list
+
+.PARAMETER Tools
+The tool[s] for which uploads are expected, as a semicolon delimited list
+
+.PARAMETER OutputFile
+The fully qualified path to the output file. Assumes that the folder already exists
+
+.Example Usage 
+.\create-tsa-config.ps1 -InstanceUrl MyInstancUrl -ProjectName MyProject -CodeBaseAdmins Domain/user1;Domain/user2 -AreaPath MyAreaPath -IterationPath -MyIterationPath -NotificationAliases MyTeam@MyCompany.com -Tools CredScan;Roslyn -OutputFolder .\tsa.config
```
Comment (dbjorge) [inline tools/scripts/pipeline/build/create-tsa-config.ps1:32]: This didn't quite run as-written. Update with quoting, fix for `-OutputFolder` -> `-OutputFile`, and using [doc format suggested by PowerShell docs](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_comment_based_help?view=powershell-7.3#comment-based-help-for-a-script):

```suggestion
.EXAMPLE
PS> .\create-tsa-config.ps1 `
        -InstanceUrl "MyInstanceUrl" `
        -ProjectName "MyProject" `
        -CodeBaseAdmins "Domain/user1;Domain/user2" `
        -AreaPath "MyAreaPath" `
        -IterationPath "MyIterationPath" `
        -NotificationAliases "MyTeam@MyCompany.com" `
        -Tools "CredScan;Roslyn" `
        -OutputFile ".\tsa.config"
```

**Review by dbjorge — APPROVED** [review]

**Pull request merged**